src/sunmatrix/band/fmod/fsunmatrix_band_mod.f90

```diff
@@ -557,4 +553,34 @@ function FSUNMatSpace_Band(a, lenrw, leniw) &
 end function
 
 
+function FSUNBandMatrix_Data(a) &
+result(swig_result)
+use, intrinsic :: ISO_C_BINDING
+real(C_DOUBLE), dimension(:), pointer :: swig_result
+type(SUNMatrix), target, intent(inout) :: a
+type(C_PTR) :: fresult 
+type(C_PTR) :: farg1 
+
+farg1 = c_loc(a)
+fresult = swigc_FSUNBandMatrix_Data(farg1)
+call c_f_pointer(fresult, swig_result, [FSUNBandMatrix_LData(a)])
+end function
+
+function FSUNBandMatrix_Column(a, j) &
+result(swig_result)
+use, intrinsic :: ISO_C_BINDING
+real(C_DOUBLE), dimension(:), pointer :: swig_result
+type(SUNMatrix), target, intent(inout) :: a
+integer(C_INT64_T), intent(in) :: j
+type(C_PTR) :: fresult 
+type(C_PTR) :: farg1 
+integer(C_INT64_T) :: farg2 
+
+farg1 = c_loc(a)
+farg2 = j
+fresult = swigc_FSUNBandMatrix_Column(farg1, farg2)
+call c_f_pointer(fresult, swig_result, [FSUNBandMatrix_UpperBandwidth(A)+FSUNBandMatrix_LowerBandwidth(a)+1])
```

Comment:
I'm confused on the indexing here. The result of `SUNBandMatrix_Column` returns a pointer to the diagonal entry and is indexed (in C) from -mu to ml but this will be indexed from 1 to mu+ml+1.

Reply:
Yeah, I am not sure what would be quite right for Fortran. In C you can index from `-mu` to `ml` and this corresponds to a range of `mu+ml+1` hence why the shape of the array returned is equal to that. However,  as far as I can figure you cannot actually index like that in Fortran without violating the bounds checks, and really since the `SUNBandMatrix_Column` function returns a pointer to the diagonal it only seems to be possible to legally access that diagonal entry... so this should probably return an array of shape [1] or we should remove the function from the Fortran interface.

Reply:
I left the function in the interface, but return the shape as `[1]` and then added a warning to the docs.